manual/query_builder/select/README.md

```diff
@@ -0,0 +1,338 @@
+
+SelectFrom selectUser = selectFrom("user");
+```
+
+Note that, at this stage, the query can't be built yet. You need at least one selector.
+
+### Selectors
+
+A selector is something that appears after the `SELECT` keyword, and will become a column in the
+result set. Its simplest form is a column identifier, but it can be a more complex expression.
+
+To create a selector, use one of the `getXxx()` methods of `QueryBuilderDsl`, and pass it to the
+`selector()` method:
+
+```java
+Selector getFirstName = getColumn("first_name");
+selectFrom("user").selector(getFirstName);
+// SELECT first_name FROM user
+```
+
+If you have multiple selectors, you can also use `selectors()` to add them all in a single call.
+This is a bit more efficient since it creates less temporary objects: 
+
+```java
+selectFrom("user").selectors(
+    getColumn("first_name"),
+    getColumn("last_name"));
+// SELECT first_name,last_name FROM user
+```
+
+Finally, there are fluent shortcuts to create and add the selector in a single call. This is
+probably the most readable if you're building the query statically:
+
+```java
+selectFrom("user")
+    .column("first_name")
+    .column("last_name");
+// SELECT first_name,last_name FROM user
+```
+
+Use an alias to give a selector a different name in the result set:
+
+```java
+selectFrom("user").selector(getColumn("first_name").as("first"));
+// SELECT first_name AS first FROM user
+```
+
+With the fluent shortcuts, chain the call after the selector:
+
+```java
+selectFrom("user").column("first_name").as("first");
+// SELECT first_name AS first FROM user
+```
+
+In addition to `column`, the query builder provides many other selectors. Some of them only work
+with newer Cassandra versions, always check what your target platform supports.
+
+#### Star selector and count
+
+`all` is the classic "star" selector that returns all columns. It cannot be aliased, and must be the
+only selector:
+
+```java
+selectFrom("user").all();
+// SELECT * FROM user
+
+selectFrom("user").all().as("everything");
+// throws IllegalStateException: Can't alias the * selector
+```
+
+If you add it to a query that already had other selectors, they will get removed:
+
+```java
+selectFrom("user").column("first_name").all();
+// SELECT * FROM user
+```
+
+If you add other selectors to a query that already had the star selector, the star selector gets
+removed:
+
+```java
+selectFrom("user").all().column("first_name");
+// SELECT first_name FROM user
+```
+
+If you add multiple selectors at once, and one of them is the star selector, an exception is thrown: 
+
+```java
+selectFrom("user").selectors(getColumn("first_name"), getAll(), getColumn("last_name"));
+// throws IllegalArgumentException: Can't pass the * selector to selectors()
+```
+
+`countAll` counts the number of rows:
+
+```java
+selectFrom("user").countAll();
+// SELECT count(*) FROM user
+```
+
+#### Arithmetic operations
```

Comment:
Is it worth indicating that these are only supported by C* 4+?

Reply:
Yes, maybe we should indicate the minimum version for every feature in the manual and javadocs. I stopped doing it at some point because it's not always trivial to track down, and a bit off-topic (you should know what query you want to write in the first place, not use the builder to find out what you can do). But it would be nice to go the extra mile anyway.